crates/block-producer/src/batch_builder/mod.rs

```diff
@@ -1,243 +1,111 @@
-use std::{
-    cmp::min,
-    collections::{BTreeMap, BTreeSet},
-    num::NonZeroUsize,
-    ops::Range,
-    sync::Arc,
-    time::Duration,
-};
+use std::{num::NonZeroUsize, ops::Range, sync::Arc, time::Duration};
 
-use miden_objects::{
-    accounts::AccountId,
-    assembly::SourceManager,
-    notes::NoteId,
-    transaction::{OutputNote, TransactionId},
-    Digest,
-};
+use miden_objects::transaction::ProvenTransaction;
 use rand::Rng;
 use tokio::{sync::Mutex, task::JoinSet, time};
-use tonic::async_trait;
 use tracing::{debug, info, instrument, Span};
 
 use crate::{
-    block_builder::BlockBuilder,
     domain::transaction::AuthenticatedTransaction,
     mempool::{BatchJobId, Mempool},
-    ProvenTransaction, SharedRwVec, COMPONENT,
+    COMPONENT, SERVER_BUILD_BATCH_FREQUENCY,
 };
 
-#[cfg(test)]
-mod tests;
+// FIXME: fix the batch builder tests.
+// #[cfg(test)]
+// mod tests;
 
 pub mod batch;
 pub use batch::TransactionBatch;
 use miden_node_utils::formatting::{format_array, format_blake3_digest};
 
-use crate::{errors::BuildBatchError, store::Store};
-
-// BATCH BUILDER
-// ================================================================================================
-
-/// Abstraction over batch proving of transactions.
-///
-/// Transactions are aggregated into batches prior to being added to blocks. This trait abstracts
-/// over this responsibility. The trait's goal is to be implementation agnostic, allowing for
-/// multiple implementations, e.g.:
-///
-/// - in-process cpu based prover
-/// - out-of-process gpu based prover
-/// - distributed prover on another machine
-#[async_trait]
-pub trait BatchBuilder: Send + Sync + 'static {
-    /// Start proving of a new batch.
-    async fn build_batch(&self, txs: Vec<ProvenTransaction>) -> Result<(), BuildBatchError>;
-}
+use crate::errors::BuildBatchError;
 
-// DEFAULT BATCH BUILDER
+// BATCH PROVER
 // ================================================================================================
 
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct DefaultBatchBuilderOptions {
-    /// The frequency at which blocks are created
-    pub block_frequency: Duration,
-
-    /// Maximum number of batches in any given block
-    pub max_batches_per_block: usize,
-}
-
-pub struct DefaultBatchBuilder<S, BB> {
-    store: Arc<S>,
-
-    block_builder: Arc<BB>,
-
-    options: DefaultBatchBuilderOptions,
-
-    /// Batches ready to be included in a block
-    ready_batches: SharedRwVec<TransactionBatch>,
+pub struct BatchProver {
```

Comment:
Similar question as in one of the previous comments: why not name this `BatchBuilder`? "Batch building" in my mind is a more general activity with "batch proving" as one of its subtasks.